ext/ldap/ldap.c

```diff
@@ -1016,6 +1016,77 @@ PHP_FUNCTION(ldap_connect)
 }
 /* }}} */
 
+#ifdef HAVE_ORALDAP
+#ifdef LDAP_API_FEATURE_X_OPENLDAP
+PHP_FUNCTION(ldap_connect_wallet) {
+	char *host = NULL;
+	size_t hostlen = 0;
+	char *wallet = NULL, *walletpasswd = NULL;
+	size_t walletlen = 0, walletpasswdlen = 0;
+	zend_long authmode = GSLC_SSL_NO_AUTH;
+	int ssl = 0;
+
+	ldap_linkdata *ld;
+	LDAP *ldap = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s!ss|l",
+		&host, &hostlen, &wallet, &walletlen, &walletpasswd, &walletpasswdlen, &authmode) != SUCCESS
+	) {
```

Comment:
Surely the ``host`` should never be null?

Reply:
I don't get the question :) It can be null based on the code:
```
char *url = host;
if (url && !ldap_is_ldap_url(url)) {
```

Reply:
My question is more, does it *ever* make sense to pass ``null`` as the host, as that seems pretty dubious to me.